client/blocks/stats-navigation/index.js

```diff
@@ -0,0 +1,97 @@
+/** @format */
+/**
+ * External Dependencies
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import SectionNav from 'components/section-nav';
+import NavItem from 'components/section-nav/item';
+import NavTabs from 'components/section-nav/tabs';
+import Intervals from './intervals';
+import FollowersCount from 'blocks/followers-count';
+import { isPluginActive } from 'state/selectors';
+import { isJetpackSite } from 'state/sites/selectors';
+import { navItems, intervals } from './constants';
+import { getJetpackSites } from 'state/selectors';
+import QueryJetpackPlugins from 'components/data/query-jetpack-plugins';
+import config from 'config';
+
+const StatsNavigation = props => {
+	const { isJetpack, isStore, slug, selectedItem, interval, jetPackSites } = props;
+	function isValidItem( item ) {
+		switch ( item ) {
+			case 'activity':
+				return config.isEnabled( 'jetpack/activity-log' ) && isJetpack;
+			case 'store':
+				return isStore;
+			default:
+				return true;
+		}
+	}
+	const { label, showIntervals, path } = navItems[ selectedItem ];
+	const slugPath = `${ slug ? '/' : '' }${ slug || '' }`;
+	const pathTemplate = `${ path }/{{ interval }}${ slugPath }`;
+	return (
+		<div className="stats-navigation">
+			<SectionNav selectedText={ label }>
+				{ isJetpack && <QueryJetpackPlugins siteIds={ jetPackSites.map( site => site.ID ) } /> }
```

Comment:
Great question, I had to do some digging for this one. When switching sites, the base path is taken from the current path. So switching sites keeps the same part of Calypso visible.

But that won't work when switching to from a store to non-store site. In order to know if a newly selected site should redirect to Site Stats instead, we'll need to know about Jetpack plugins for all sites.

https://github.com/Automattic/wp-calypso/blob/1ff6fe48a4361b409cbd04a0a9bb1e170920caa0/client/components/site-selector/index.jsx#L473-L478

Reply:
I tried using `siteIds={ [ siteId ] }` and saw what appeared to be the same behavior. We can save `jetpackSites - 1` requests, but there may be more complexity here than I'm appreciating.

I tried clearing mystate, refreshing on my Jetpack store site, then navigating to another Jetpack site with no store. I was redirected to `/stats/day`.

Here's the navigation with 1 site request (after clearing application state):

![nav-store-nostore](https://user-images.githubusercontent.com/841763/30960257-ca921cfc-a442-11e7-8d36-f8a1eb2063ce.gif)

These gifs aren't so great, but one is all sites, which requests plugins for all sites when you navigate between sites. The other requests 1 site for each navigation between sites.

![reqs-allsites](https://user-images.githubusercontent.com/841763/30960464-5bddc8e6-a443-11e7-8674-b10455b65fd5.gif)

![reqs-1site](https://user-images.githubusercontent.com/841763/30960467-5dda36e8-a443-11e7-87f0-542c40b1bdd1.gif)

There may be nuances or complexity here I don't appreciate, so do as you see fit 🙂

Reply:
I also used `siteIds={ [ siteId ] }` to confirm. The unexpected behaviour happens when you switch from one store to another. Since only the first store's plugins have been queried, it won't know the second store has Woo installed and will take the user to Site Stats.

Its a bit edge case.

Reply:
Right! I get it now. 👍 

Thanks for explaining 🙂